### PR TITLE
Add method to fetch specials for customers

### PR DESCRIPTION
**Pull request by lawyu89**
Add /v4.1/customer/specials (lists specials) support to Booker client [ADO#716208](https://mindbody.visualstudio.com/Marketing%20Products/_sprints/taskboard/Babel%20Fish/Marketing%20Products/Sprint%202020-11?workitem=716208)

### CODE REVIEW

File: lib/booker/v4/models/special.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+module Booker
+  module V4
+    module Models
+      class Special < Model
+        attr_accessor 'ID',
```
Comment (clacan) [inline lib/booker/v4/models/special.rb:5]: I think attributes that are objects need to have nested classes, e.g., 'SpecialItems', 'Brands', 'FreeItems'

**Review by clacan — COMMENTED** [review]
You need classes and specs for BookingDateType, BookingEarlyBirdUnitType, BookingLastMinuteUnitType, Price Range, PriceRangeScope, Exclusion. I believe the only custom class will be PriceRange, and the others will just inherit from Type.

**Review by clacan — APPROVED** [review]

File: spec/lib/booker/v4.1/customer_spec.rb [hunk 2]
```diff
@@ -273,6 +274,40 @@
...
+    context 'other params' do
+      let(:expected_params) {described_class::DEFAULT_PAGINATION_PARAMS.merge({
+                                                                                access_token: 'access_token',
+                                                                                LocationID: booker_location_id,
+                                                                                another_option: 'foo'
+                                                                              })}
+
+      it 'merges passed in params into base params' do
+        expect(client.specials(location_id: booker_location_id, params: {another_option: 'foo'})).to eq []
+      end
```
Comment (otheogar) [inline spec/lib/booker/v4.1/customer_spec.rb:307]: shouldn't this be `expected_params`?

Comment (otheogar) [reply spec/lib/booker/v4.1/customer_spec.rb:307]: nvm I see what's happening. I find it a little difficult to read when the expect is in the before block. It's not very clear what the test is testing based on the actual lines in the test. I know this is a pattern we use though so it's fine, just confused me.

**Review by otheogar — APPROVED** [review]
LGTM

**Pull request merged**